File: Frontend/src/proxy/Interfaces/EntityDtos/audited-entity-dto.ts
import { EntityDto } from "./entity-dto";

export interface AuditedEntityDto<T> extends EntityDto<number> {
    creationTime: Date,
    creatorId: T,
    creatorName: string,
    IsDeleted: boolean,
    DeletionTime: Date,
    DeleterId: T,
    DeleterName: string
}

File: Backend/IronMan.Core/Dtos/EntityDtos/AuditedEntityDto.cs
namespace IronMan.Core.Dtos.EntityDtos
{
    public class AuditedEntityDto<T> : EntityDto<T>
    {
        public DateTime? CreationTime { get; set; }
        public int? CreatorId { get; set; }
        public string? CreatorName { get; set; }
        public bool? IsDeleted { get; set; }
        public DateTime? DeletionTime { get; set; }
        public int? DeleterId { get; set; }
        public string? DeleterName { get; set; }
    }
}
